youtube_dl/extractor/ximalaya.py

```diff
@@ -0,0 +1,174 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+import itertools
+import re
+
+from .common import InfoExtractor
+
+
+class XimalayaBaseIE(InfoExtractor):
+    _GEO_COUNTRIES = ['CN']
+
+
+class XimalayaIE(XimalayaBaseIE):
+    IE_NAME = 'ximalaya'
+    IE_DESC = 'ximalaya.com'
+    _VALID_URL = r'https?://(?:www\.|m\.)?ximalaya\.com/(?P<uid>[0-9]+)/sound/(?P<id>[0-9]+)/?'
+    _USER_URL_FORMAT = 'http://www.ximalaya.com/zhubo/%i/'
+    _TESTS = [
+        {
+            'url': 'http://www.ximalaya.com/61425525/sound/47740352/',
+            'info_dict': {
+                'id': '47740352',
+                'ext': 'm4a',
+                'title': '261.唐诗三百首.卷八.送孟浩然之广陵.李白',
+                'description': 'contains:孤帆远影碧空尽，惟见长江天际流。',
+                'uploader': '小彬彬爱听书',
+                'uploader_id': 61425525,
+                'view_count': int,
+                'like_count': int,
+            }
+        },
+        {
+            'url': 'http://m.ximalaya.com/61425525/sound/47740352/',
+            'info_dict': {
+                'id': '47740352',
+                'ext': 'm4a',
+                'title': '261.唐诗三百首.卷八.送孟浩然之广陵.李白',
+                'description': 'contains:孤帆远影碧空尽，惟见长江天际流。',
+                'uploader': '小彬彬爱听书',
+                'uploader_id': 61425525,
+            }
+        },
+    ]
+
+    def _real_extract(self, url):
+
+        is_m = 'm.ximalaya' in url
+
+        audio_id = self._match_id(url)
+        webpage = self._download_webpage(url, audio_id,
+                                         note='Download sound page for %s' % audio_id,
+                                         errnote='Unable to get sound page')
+
+        audio_info_file = 'http://m.ximalaya.com/tracks/%s.json' % audio_id
+        audio_info = self._download_json(audio_info_file, audio_id,
+                                         'Downloading info json %s' % audio_info_file,
+                                         'Unable to download info file')
+
+        formats = []
+        for bps, k in (('24k', 'play_path_32'), ('64k', 'play_path_64')):
+            if audio_info.get(k):
+                formats.append({
+                    'format_id': bps,
+                    'url': audio_info[k],
+                    'ext': 'm4a',
```

Comment:
the 'ext' field is needed when test, otherwise I got this error :
Exception: Test definition incorrect. The output file cannot be known. Are both 'id' and 'ext' keys present?